feat(docs): adds LCM docs

uwe-mayer opened the pull request:
## Description
Adds documentation on how to do lifecycle management with the Plugin machinery of Greenhouse.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

uwe-mayer committed:
feat(docs): adds LCM docs

ibakshay reviewed (changes requested):
Left some minor comments :)

uwe-mayer committed:
Update docs/user-guides/plugin/lifecycle.md

Co-authored-by: Akshay Iyyadurai Balasundaram <akshay.iyyadurai.balasundaram@sap.com>

uwe-mayer committed:
Update docs/user-guides/plugin/lifecycle.md

Co-authored-by: Akshay Iyyadurai Balasundaram <akshay.iyyadurai.balasundaram@sap.com>

uwe-mayer committed:
Update docs/user-guides/plugin/lifecycle.md

Co-authored-by: Akshay Iyyadurai Balasundaram <akshay.iyyadurai.balasundaram@sap.com>

uwe-mayer committed:
Update docs/user-guides/plugin/lifecycle.md

Co-authored-by: Akshay Iyyadurai Balasundaram <akshay.iyyadurai.balasundaram@sap.com>

uwe-mayer committed:
docs(plugin): update lifecycle docs to reflect current API

- Update version pinning status to 🟩 (implemented via Catalog git refs)
- Fix broken API reference links (missing # separator)
- Add Flux GitRepository/Kustomization explanation for Catalogs
- Update frontend reference from React components to Juno Applications
- Remove outdated 'in development' section about Version field
- Add clear explanation of branch/tag/SHA pinning strategies
- Fix YAML examples to not use Helm template syntax
- Remove TODO HTML comments
- Fix typos and grammar

abhijith-darshan commented on docs/user-guides/plugin/lifecycle.md:
```diff
@@ -0,0 +1,210 @@
+
+- **Branch tracking** (e.g. `branch: main`): You always get the latest PluginDefinitions as they are committed to the branch. This is the default behavior and enables automatic updates.
+- **Tag pinning** (e.g. `tag: "kube-monitoring/1.2.3"`): You only get PluginDefinitions as of that tagged commit. Updates only happen when you change the tag reference.
+- **SHA pinning** (e.g. `sha: "a1b2c3d4e5f6..."`): You freeze to an exact commit. This provides the strongest version guarantee.
```
all these are documented in Catalog. If something is missing we should enhance it there...

I would just give jump off points https://cloudoperators.github.io/greenhouse/docs/reference/components/catalog/#ref ??

This way if Catalog changes then documentation needs to be updated only in one place.

WDYT?

uwe-mayer replied:
done

abhijith-darshan commented on docs/user-guides/plugin/lifecycle.md:
````diff
@@ -0,0 +1,210 @@
+      alias: ingress-nginx-custom
+    - name: kube-monitoring
+      alias: kube-monitoring-custom
+```
````
These are already documented with examples in Catalog...

We should do a single jump off point to - https://cloudoperators.github.io/greenhouse/docs/reference/components/catalog/#configuring-overrides-for-plugindefinitions

uwe-mayer replied:
done

abhijith-darshan commented on docs/user-guides/plugin/lifecycle.md:
````diff
@@ -0,0 +1,210 @@
+      ref:
+        sha: "a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6a1b2"
+    path: plugindefinitions
+```
````
The individual examples are missing in Catalog documentation for this. I think we should move this to catalog docs and give jump off points

uwe-mayer replied:
done

uwe-mayer committed:
Merge branch 'main' into docs/catalogs

uwe-mayer committed:
PR remarks
